cli/dts/lib.deno.unstable.d.ts

```diff
@@ -122,7 +122,8 @@ declare namespace Deno {
 
   /** A foreign function as defined by its parameter and result types */
   export interface ForeignFunction {
-    parameters: NativeType[];
+    parameters:
```

Comment:
I would perhaps go with:
```ts
interface CallbackParameter extends Omit<ForeignFunction, "nonblocking"> {
  type: "function";
}

type ForeignFunctionParameter = NativeType | CallbackParameter;

interface ForeignFunction {
  parameters: ForeignFunctionParameter[];
}
```

ie. Avoid the extra `{ function: {}}` object wrap, and instead detect the parameter type directly from the object's type field.

Though, I don't know if this will cause some unforeseen issues in the Rust binding.